python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -140,6 +154,42 @@ def __init__(self, iface, db, parent=None):
 
         self.presetName.textChanged.connect(self.nameChanged)
 
+    def insertQueryInEditor(self, item):
+        sql = item.data(Qt.DisplayRole)
+        self.editSql.insertText(sql)
+
+    def showHideQueryHistory(self, visible):
+        self.queryHistoryWidget.setVisible(visible)
+
+    def populateQueryHistory(self):
+        self.queryHistoryTableWidget.clearContents()
+        self.queryHistoryTableWidget.setRowCount(0)
+        dictlist = self.history[self.connectionName]
+
+        if not dictlist:
+            return
+
+        for i in range(len(dictlist)):
+            self.queryHistoryTableWidget.insertRow(0)
+            queryItem = QTableWidgetItem(dictlist[i]['query'])
+            rowsItem = QTableWidgetItem(str(dictlist[i]['rows']))
+            durationItem = QTableWidgetItem(str(dictlist[i]['secs']))
+            self.queryHistoryTableWidget.setItem(0, 0, queryItem)
+            self.queryHistoryTableWidget.setItem(0, 1, rowsItem)
+            self.queryHistoryTableWidget.setItem(0, 2, durationItem)
+
+        self.queryHistoryTableWidget.resizeColumnsToContents()
+        self.queryHistoryTableWidget.resizeRowsToContents()
+
+    def writeQueryHistory(self, sql, affectedRows, secs):
```

Comment:
any chance to clear the queryHistory? it would grow in QgsSetting for the eternity :)

Reply:
could be a solution @Gustry